File: app/ros_sims/sample_sim.jl
#!/usr/bin/env julia
# should be set and run as an executable

# this file is just used for testing ROSInterface.jl

using Pkg
Pkg.activate(Base.source_dir() * "/..")

using RobotOS
@rosimport std_msgs.msg: Float32
rostypegen()
using .std_msgs.msg

function loop(pubs)
    loop_rate = Rate(2.0)
    while !is_shutdown()
        for pub in pubs
            publish(pub, Float32Msg(rand()))
        end
        rossleep(loop_rate)
    end
end

function main()
    init_node("sample_sim")
    pubs = [Publisher{Float32Msg}("value1", queue_size=1),
            Publisher{Float32Msg}("value2", queue_size=1)]
    loop(pubs)
end

if !isinteractive()
    main()
end

File: app/ros_sims/sample_sim.py
#!/usr/bin/env python
# should be set and run as an executable

# this file is just used for testing ROSInterface.jl

import rospy
from std_msgs.msg import Float32
from random import random

def loop(pubs):
    loop_rate = rospy.Rate(2.0)
    while not rospy.is_shutdown():
        for pub in pubs:
            pub.publish(Float32(random()))
        loop_rate.sleep()

def main():
    rospy.init_node("sample_sim")
    pubs = [rospy.Publisher("value1", Float32, queue_size=1),
            rospy.Publisher("value2", Float32, queue_size=1)]
    loop(pubs)

if __name__ == '__main__':
    main()
